async-nats/examples/jetstream_pull.rs

```diff
@@ -35,8 +35,10 @@ async fn main() -> Result<(), async_nats::Error> {
 
     // Publish a few messages for the example.
     for i in 0..10 {
-        jetstream
+        // await for a publish ack
+        let _ack = jetstream
             .publish(format!("events.{i}"), "data".into())
+            .await?
             .await?;
```

Comment:
```suggestion
             // Second `await` awaits publish acknowledgement.
             // This can be skipped (for the cost of processing guarantee)
             // or deferred to not block another `publish`.
            .await?;
```